### ISSUE TITLE
Spaces at beginning/end of vobsub lines are ignored

### ISSUE THREAD

**Issue by vertigo220** [SubtitleEdit/subtitleedit#3079]
I'm trying to export some lines into vobsub so I can use a black background to overlay burned-in subs, but some of the lines are shorter in the srt/sub than the hardcoded ones are. I tried stretching the black box, but there's no explicit option for doing this (it would be nice to be able to stretch it horizontally and vertically), and adding spaces at the beginning and end of the lines doesn't work, because they're just ignored during the export.

**Comment by vertigo220** [comment]
For anyone else looking for a solution to this, I found a workaround. I used Aegisub Advanced Subtitle Editor, though I think the same can be done in SubtitleEdit, I just figured it out while trying Aegisub to see if it would "see" leading and trailing spaces or not. It turns out it doesn't either, but it apparently only works in .ass format, and I knew from a previous issue here that .ass/.ssa is capable of black backgrounds like vobsub. I created a new style with a background and tweaked it to get it to look right, then figured out that you have to do a \N (which means new line) at the beginning/end of the line, with an Alt+255 character right next to it. Leaving out either will cause the black box to only extend just past the first/last word. For example, if I take a line that says "Hello world" and want it to cover a longer line, I have to enter this as the text (replace {Alt+255} with the actual keystrokes which, if you don't know, you have to hold down the Alt key while typing on the number pad, NOT the number row, or, assuming it doesn't break it, you can copy it here between the brackets [ ] and paste it in):

\N{Alt+255}        Hello world        {Alt+255}\N

Of course, use as many spaces as needed. As long as nothing is on the other side of the \N's, it won't actually create additional lines. And if it's a multi-line subtitle, just do it like this:

\N{Alt+255}        Hello        {Alt+255}\N{Alt+255}        World        {Alt+255}\N

As I said, I think the same can be done in SubtitleEdit, by saving as .ssa or .ass (.ass seems better, as it appears to support customizing how far the black background extends, whereas I don't think .ssa does). One of the problems was I was looking in the export menu for SSA/ASS, but for some reason even though vobsub is there those aren't, and they are accessed via the save as dialog. The problem is, as far as I can tell SubtitleEdit offers no way to customize the background and other options, whereas Aegisub does. It very well might, but I'll leave that to others to figure out if they want; I've spent far too much time on this already. I just hope this helps someone.

I still think it would be great if this ability could be added to the program, by just allowing the background to be resized, but it's admittedly probably rarely needed, and I suspect vobsub/ASS/SSA don't have a built-in way of handling that. So this might be the best it's going to be.

**Comment by OmrSi** [comment]
Or you can just add \h as much as you want before and after, so "Hello World" would be "\hHello World\h".
\h: Hard space in ass format.

**Comment by vertigo220** [comment]
Ah, good to know. Where were you earlier? :P  I doubt I'll redo it, but if I do I might just change the \N{Alt+255} to \h and leave the spaces as spaces, since I assume they would still register as long as they're between the \h and the actual words, and it would look a lot cleaner than \h\h\h\h\hHello World\h\h\h\h\h. Thanks for the tip!

**Comment by OmrSi** [comment]
I didn't see this until now. 😅
The same thing happened with me once and I used \h, a lot of it,  your idea to use only one is cool. :) 
I don't think there is an option to strech the black box in Vobsub and I don't think the file's format has a tag or anything for this. 
Using .ass files is better in this case. 👌😁
I had to use it once to cover a hardsub in a file, like I said, I used position tags and "\h", the result was kinda good. 

Maybe this can be closed now?
